src/multiformats/hash.cljc

```diff
@@ -101,101 +101,127 @@
 
 
 ;; ## Multihash Type
+#?(:bb
+   (do
+     (defn length [mhash]
```

Comment:
Is this all necessary because babashka doesn't have `ILookup` or a way of implementing custom keyword lookup semantics on records? I wouldn't mind making these accessor functions a standard part of the namespace to cut down on differences, though it does introduce a bunch of shadowing issues since these symbols are also often used for local bindings (as comes up in the test ns).

I also wonder if the bb implementation could just set the appropriate k/v pairs on the record at construction time. They wouldn't be immutable I guess, but the interface would stay the same then.

Reply:
Babashka currently does not support implementing `ILookup` on `deftype` - `deftype` is one of the least supported things in bb and the main cause of incompatibilities. The reason for this is that in a native image you can't create new types from scratch, so you would have to pre-create a type for every combination of interface you want to implement.

It is possible to create a custom map type in babashka using `proxy` + `clojure.lang.APersistentMap clojure.lang.IMeta clojure.lang.IObj` since that's a predefined option:

https://blog.wsscode.com/guide-to-custom-map-types/#cmt-in-babashka

> I also wonder if the bb implementation could just set the appropriate k/v pairs on the record at construction time.

That would also be an option, we'd do some work ahead of time rather than lazily like it's happening now in the JVM implementation.

Reply:
Maybe the `proxy` solution would be the closest to the original, since you would also be able to implement equality.